anaBenn/Homework_#2.js

```diff
@@ -0,0 +1,533 @@
+//which is an array holding all the values to && together. If the argument you provide when calling the function 
+//holds values [a,b,c...z], as in andN([a,b,c...z]), the function should return the same result as the expression 
+//(a && b && c && ... z). Make sure to handle two special cases: length 0 (then return true) and length 1 (then return 
+//  that single value). Again, don't use &&.
+
+values = ["banana", 25, true, "mickey mouse", false, 1999];
+
+function andN(values) {
+  var n = values.length;
+
+  if (n == 0) {
+    return true;
+  }
+  if (n == 1) {
+    return values[0];
+  }
+  for (var i = 0; i < n; i++) {
+    if (!values[i]) {
+      return values[i];
+    }
+  }
+  return values[i - 1];
+}
+console.log(andN(values));
+
+
+//NOTE: (Logical AND) Returns expr1 if it can be converted to false; otherwise, returns expr2. Thus, when used with 
+//Boolean values, && returns true if both operands are true; otherwise, returns false.
+
+//d) You've just realized that your effort was doomed: neither of your functions can replace the && operator
+//in certain circumstances. Explain why, and find an example which demonstrates failure.
+
+
+Computers are big ol' dummies (so much so, that my computer just turn "dummies" (and everything that is in these parens) into a string...
+let me fix that').  There are certain instances that && is needed in order for the computer to understand the data that it is 
+being fed and the instructions that we are giving it. These instances include, but in no way are limited to:
+
+-If you are instructing a computer to complete a task or run a method/function only after another task has been completed, && is the way to do this.  
+You cannot use ||, "and"or a ,.  For some situations mentioed above, you could use an if/else statement to satisfy these conditions, but the code would 
+not be as eloquent. 
+
+-Both && and || are short-circuit evaluations - This short-circuiting is great for performance, as it allows significant bits of calculations to be skipped. 
+
+-Both && and || only evaluate to their last evaluated operator.
+
+
+//5) (Moderate, 35%)
+
+//Imagine that a deck of playing cards is sorted by rank and suit: first all the Aces, then the Twos, etc, with 
+//the Kings last. Within each rank, the suits are in the order Hearts, Diamonds, Spades, Clubs. Number each card 
+//in order from 0 to 51 (i.e. 0=Ace of Hearts; 1=Ace of Diamonds; 51=King of Clubs), and let that ID number represent 
+//the corresponding card. Use this encoding scheme for each part below.
+
+function rank(card) {
+  return Math.floor(card / 4) + 1;
+}
+
+function suit(card) {
+  var remainder = card % 4;
+  return remainder + 1;
+}
+
+function cardID(rank, suit) {
+  var cardRank = (rank - 1) * 4;
+  var cardSuit = (suit - 1);
+  return cardRank + cardSuit;
+}
+
+function color(id) {
+  var redOrBlack = (id % 4) + 1;
```

Comment:
This working 99.9% the way it should, except that your returning 'black' and 'red' should be swapped around. Right now the color of the ace of diamonds is returning 'black', for example. A quick glance below shows you've corrected this for your validity checked code, good catch..